src/main/java/org/springframework/social/partnercenter/api/agreement/AgreementTemplate.java

```diff
@@ -0,0 +1,42 @@
+package org.springframework.social.partnercenter.api.agreement;
+
+import static org.springframework.social.partnercenter.api.validation.Assertion.notNull;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.ResponseEntity;
+import org.springframework.social.partnercenter.api.PagingResourceTemplate;
+import org.springframework.social.partnercenter.api.PartnerCenterResponse;
+import org.springframework.social.partnercenter.http.client.RestResource;
+
+public class AgreementTemplate extends PagingResourceTemplate<AgreementMetaData> implements AgreementOperations {
+	private RestResource restResource;
+
+	public AgreementTemplate(RestResource restResource, boolean isAuthorized) {
+		super(restResource, isAuthorized, new ParameterizedTypeReference<PartnerCenterResponse<AgreementMetaData>>() {});
+		this.restResource = restResource;
+	}
+
+	@Override
+	public ResponseEntity<PartnerCenterResponse<AgreementMetaData>> getAgreementMetadatas() {
```

Comment:
only one question on the PR @christophele-appdirect 

`getAgreementMetadatas` and `getConfirmations` are named plural, do they effectively return a list of objects?

Reply:
yes they are!

hint: when a response is typed with ParterCenterResource, it means the response is paginated!